# Data Understanding

El objetivo de este notebook es comprender la estructura del dataset Superstore, identificar las variables disponibles y detectar posibles problemas de calidad de datos antes de iniciar la fase de limpieza.

Carga de datos

In [1]:
import pandas as pd

df = pd.read_csv("../data/raw/Sample - Superstore.csv", encoding="latin1")

df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


El dataset se carga desde la carpeta `data/raw`.  
Se utiliza `encoding="latin1"` porque el archivo presenta una codificación diferente a UTF-8.

Tamaño del dataset

In [2]:
df.shape

(9994, 21)

El dataset contiene 9.994 registros y 21 variables.

Información general

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

La mayoría de variables son categóricas, aunque también existen variables numéricas como Sales, Quantity, Discount y Profit.

No se observan valores nulos en las columnas del dataset.

Estadísticas descriptivas

In [4]:
df.describe()

,Row ID,Postal Code,Sales,Quantity,Discount,Profit
count,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000
mean,4997.500000,55190.379428,229.858001,3.789574,0.156203,28.656896
std,2885.163629,32063.693350,623.245101,2.225110,0.206452,234.260108
min,1.000000,1040.000000,0.444000,1.000000,0.000000,-6599.978000
25%,2499.250000,23223.000000,17.280000,2.000000,0.000000,1.728750
50%,4997.500000,56430.500000,54.490000,3.000000,0.200000,8.666500
75%,7495.750000,90008.000000,209.940000,5.000000,0.200000,29.364000
max,9994.000000,99301.000000,22638.480000,14.000000,0.800000,8399.976000


Las estadísticas descriptivas permiten observar el comportamiento general de las variables numéricas.

Se observa que Profit incluye valores negativos, lo que indica que algunos pedidos generan pérdidas.

Revisión de duplicados

In [5]:
df.duplicated().sum()

np.int64(0)

No se encontraron registros duplicados en el dataset.

Revisión de categorías principales

In [6]:
df["Category"].value_counts()

Category
Office Supplies    6026
Furniture          2121
Technology         1847
Name: count, dtype: int64

In [7]:
df["Region"].value_counts()

Region
West       3203
East       2848
Central    2323
South      1620
Name: count, dtype: int64

In [8]:
df["Segment"].value_counts()

Segment
Consumer       5191
Corporate      3020
Home Office    1783
Name: count, dtype: int64

El dataset contiene tres categorías principales de productos: Office Supplies, Furniture y Technology.

También incluye cuatro regiones de venta y tres segmentos de clientes.

Valores únicos

In [9]:
df.nunique()

Row ID           9994
Order ID         5009
Order Date       1237
Ship Date        1334
Ship Mode           4
Customer ID       793
Customer Name     793
Segment             3
Country             1
City              531
State              49
Postal Code       631
Region              4
Product ID       1862
Category            3
Sub-Category       17
Product Name     1850
Sales            5825
Quantity           14
Discount           12
Profit           7287
dtype: int64

El dataset contiene 793 clientes únicos, 1.850 productos únicos y 5.009 pedidos únicos.

Esto permite realizar análisis desde diferentes perspectivas: clientes, productos, pedidos y regiones.

Ventas y beneficios totales

In [10]:
df["Sales"].sum()

np.float64(2297200.8603)

In [11]:
df["Profit"].sum()

np.float64(286397.0217)

El dataset registra unas ventas totales aproximadas de 2.297.200 dólares y un beneficio total aproximado de 286.397 dólares.

Primer análisis por categoría

In [12]:
df.groupby("Category")["Sales"].sum().sort_values(ascending=False)

Category
Technology         836154.0330
Furniture          741999.7953
Office Supplies    719047.0320
Name: Sales, dtype: float64

In [13]:
df.groupby("Category")["Profit"].sum().sort_values(ascending=False)

Category
Technology         145454.9481
Office Supplies    122490.8008
Furniture           18451.2728
Name: Profit, dtype: float64

Technology es la categoría con mayores ventas y beneficios.

Furniture presenta un volumen de ventas elevado, pero su beneficio es considerablemente menor en comparación con Technology y Office Supplies.

## Conclusión

El dataset Superstore contiene información suficiente para realizar un análisis de negocio completo.

No presenta valores nulos ni duplicados, por lo que se encuentra en buen estado para continuar con la fase de limpieza y preparación.

Los primeros análisis muestran que Technology es la categoría con mejores resultados, mientras que Furniture podría presentar problemas de rentabilidad que serán investigados en el análisis de negocio.